### ISSUE TITLE
[javascript] problem with continue statement and newline followed by whitespace

### ISSUE THREAD

**Issue by meraedit** [antlr/grammars-v4#3772]
```js
for (var i = 0; i < [].length; i++) {
	var o = {}
	if (false) continue
	o[test]   //	-  'o' is label for the continue statement, which is wrong
		  //	- no viable alternative at input '['
}
```

The result is 

```
continue o;
[test]
```

instead of

```
continue;
o[test]
```

The grammar contains this rule, `Continue ({this.notLineTerminator()}? identifier)? eos`, therefore `o` should not be a label of continue because there is a newline between them.

The same applies to `break` and `return` statements.

**Comment by bkiers** [comment]
It appears to be an issue with the `notLineTerminator()` method: https://github.com/antlr/grammars-v4/issues/3777

**Comment by bkiers** [comment]
Did a couple of tests:

## JS 1
```js
{
  continue
}
```

## Tree 1
```
'- block
   |- { (OpenBrace)
   |- statementList
   |  '- statement
   |     '- continueStatement
   |        |- continue (Continue)
   |        '- eos
   '- } (CloseBrace)
```

## JS 2
{
  continue a
}
```

## Tree 2
'- block
   |- { (OpenBrace)
   |- statementList
   |  '- statement
   |     '- continueStatement
   |        |- continue (Continue)
   |        |- identifier
   |        |  '- a (Identifier)
   |        '- eos
   '- } (CloseBrace)
```

## JS 3
{
  continue
  a
}
```

## Tree 3
'- block
   |- { (OpenBrace)
   |- statementList
   |  '- statement
   |     '- continueStatement
   |        |- continue (Continue)
   |        |- identifier
   |        |  '- a (Identifier)
   |        '- eos
   '- } (CloseBrace)
```

## JS 4
{
  continue
  a.mu()
}
```

## Tree 4
'- block
   |- { (OpenBrace)
   |- statementList
   |  |- statement
   |  |  '- continueStatement
   |  |     |- continue (Continue)
   |  |     '- eos
   |  '- statement
   |     '- expressionStatement
   |        |- expressionSequence
   |        |  '- argumentsExpression
   |        |     |- memberDotExpression
   |        |     |  |- identifierExpression
   |        |     |  |  '- identifier
   |        |     |  |     '- a (Identifier)
   |        |     |  |- . (Dot)
   |        |     |  '- identifierName
   |        |     |     '- identifier
   |        |     |        '- mu (Identifier)
   |        |     '- arguments
   |        |        |- ( (OpenParen)
   |        |        '- ) (CloseParen)
   |        '- eos
   '- } (CloseBrace)
```

So the result from JS test 3 is wrong, the `a` should be treated as a separate `statement`. oddly enough, JS test 4 goes correct...

If I change `here` in the `JavaScriptParserBase` into this:

```java
private boolean here(final int type) {

    Token currentToken = _input.LT(-1);
    Token nextToken = _input.get(currentToken.getTokenIndex() + 1);

    return (nextToken.getChannel() == Lexer.HIDDEN) && (nextToken.getType() == type);
}
```

all tests go as expected.

I will do some more tests and will make a PR if all goes well.

**Comment by meraedit** [comment]
Hello, I was away for a few days and I see that #3778 
is a better fix for this problem that the PR #3773 which I provided 
So #3773 should not have been merged into master...
Thanks @bkiers

**Comment by meraedit** [comment]
Unfortunately the fix does not work for the situation when continue is followed by comment (single or multiline).
I tried to change the notLineTerminator method to also check for JavaScriptParser.MultiLineComment, JavaScriptParser.SingleLineComment, whitespaces and combinations of whitespace + comment.
But I couldn't manage to make it match all the situations that are now described in [examples/Continue.js](https://github.com/Servoy/grammars-v4/blob/continue_line_terminators_comments/javascript/javascript/examples/Continue.js)
and it seems that notLineTerminator should actually be !lineTerminatorAhead
